File: Estudos/Busca Binária/ondeEstaOMarmore.py
import sys

input = sys.stdin.readline  # Substitui a função input

n = 1
q = 1
while(n!=0 and q!=0 ):
    n,q = map(int,input().split())
    a = []

    for i in range(n):
        b = int(input())
        a.append(b)
    a.sort()
    for i in range(q):
        numAchar = int(input())
        l = 0
        r = len(a)-1
        res = 0
        while(r>=l):
            mid= (l+r)//2
            if a[mid] == numAchar:
                res = mid
                r = mid-1
            elif numAchar > a[mid] :
                l = mid+1
            else:
                r=mid-1
        print(f"CASE# {q}:")
    
        if res == 0:
            print(f"{numAchar} not found")
        else:
            print(f"{numAchar} found at {res+1}")

File: Estudos/Busca Binária/ondeEstaOMarmore.cpp
/* https://judge.beecrowd.com/pt/problems/view/1025 */
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int caso = 1;
    int n, q;

    while (cin >> n >> q) {
        if (n == 0 && q == 0)
            break;

        vector<int> a(n);
        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }

        sort(a.begin(), a.end());

        cout << "CASE# " << caso << ":" << endl;

        for (int i = 0; i < q; i++) {
            int achar;
            cin >> achar;

            int l = 0, r = n - 1;
            int res = -1;

            while (l <= r) {
                int mid = (l + r) / 2;
                if (a[mid] == achar) {
                    res = mid;
                    r = mid - 1;
                } else if (achar > a[mid]) {
                    l = mid + 1;
                } else {
                    r = mid - 1;
                }
            }

            if (res == -1)
                cout << achar << " not found" << endl;
            else
                cout << achar << " found at " << res + 1 << endl;
        }

        caso++;
    }

    return 0;
}
